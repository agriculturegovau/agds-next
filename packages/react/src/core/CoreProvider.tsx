import {
	forwardRef,
	createContext,
	useContext,
	PropsWithChildren,
	AnchorHTMLAttributes,
} from 'react';

const DefaultLinkComponent = forwardRef<
	HTMLAnchorElement,
	// Using `any` here because the href prop differs between routing frameworks
	// eslint-disable-next-line @typescript-eslint/no-explicit-any
	LinkProps & { href?: any }
>(function DefaultLinkComponent(props, ref) {
	return <a ref={ref} {...props} />;
});

export type NativeLinkProps = AnchorHTMLAttributes<HTMLAnchorElement>;

export type LinkProps = PropsWithChildren<{
	/** Describes the anchor element to assistive technologies. */
	'aria-label'?: string;
	/** The CSS class name, typically generated from the `css` prop. */
	className?: string;
	/** Causes the browser to treat the linked URL as a download. */
	download?: NativeLinkProps['download'];
	/** The URL that the hyperlink points to. */
	href?: string;
	/** The ID of the hyperlink. */
	id?: string;
	onClick?: NativeLinkProps['onClick'];
	onMouseEnter?: NativeLinkProps['onMouseEnter'];
	onMouseLeave?: NativeLinkProps['onMouseLeave'];
	onMouseOut?: NativeLinkProps['onMouseOut'];
	onMouseOver?: NativeLinkProps['onMouseOver'];
	/** How much of the referrer to send when following the link. */
	referrerPolicy?: NativeLinkProps['referrerPolicy'];
	/** The relationship of the linked URL as space-separated link types. */
	rel?: string;
	/** WAI-ARIA role. */
	role?: NativeLinkProps['role'];
	/** Where to display the linked URL, as the name for a browsing context (a tab, window, or <iframe>). */
	target?: NativeLinkProps['target'];
}>;

export const coreContext = createContext({
	linkComponent: DefaultLinkComponent,
});

export type LinkComponent = typeof DefaultLinkComponent;

export type CoreProviderProps = PropsWithChildren<{
	/** The global link component which can be configured for different routing libraries. Defaults to standard `a` tag. */
	linkComponent?: LinkComponent;
}>;

export function CoreProvider({ linkComponent, children }: CoreProviderProps) {
	return (
		<coreContext.Provider
			value={{
				linkComponent: linkComponent ?? DefaultLinkComponent,
			}}
		>
			{children}
		</coreContext.Provider>
	);
}

export function useLinkComponent() {
	const { linkComponent } = useContext(coreContext);
	return linkComponent;
}
