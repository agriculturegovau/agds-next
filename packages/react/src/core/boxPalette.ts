import { css } from '@emotion/react';
import { RefObject, useEffect, useState } from 'react';
import { themeVars } from './theme';

// Box Palette

const boxPaletteVars = {
	palette: '--agds-palette',
	foregroundText: '--agds-foreground-text',
	foregroundAction: '--agds-foreground-action',
	foregroundMuted: '--agds-foreground-muted',
	backgroundBody: '--agds-background-body',
	backgroundShade: '--agds-background-shade',
	backgroundBodyAlt: '--agds-background-body-alt',
	backgroundShadeAlt: '--agds-background-shade-alt',
	border: '--agds-border',
	borderMuted: '--agds-border-muted',
	accent: '--agds-accent',
	focus: '--agds-foreground-focus',
	overlay: '--agds-overlay',
	systemSuccess: '--agds-system-success',
	systemSuccessMuted: '--agds-system-success-muted',
	systemWarning: '--agds-system-warning',
	systemWarningMuted: '--agds-system-warning-muted',
	systemInfo: '--agds-system-info',
	systemInfoMuted: '--agds-system-info-muted',
	systemError: '--agds-system-error',
	systemErrorMuted: '--agds-system-error-muted',
};

export const boxPalettes = {
	light: css({
		[boxPaletteVars.palette]: 'light',
		[boxPaletteVars.foregroundText]: `var(${themeVars.lightForegroundText})`,
		[boxPaletteVars.foregroundAction]: `var(${themeVars.lightForegroundAction})`,
		[boxPaletteVars.foregroundMuted]: `var(${themeVars.lightForegroundMuted})`,
		[boxPaletteVars.backgroundBody]: `var(${themeVars.lightBackgroundBody})`,
		[boxPaletteVars.backgroundShade]: `var(${themeVars.lightBackgroundShade})`,
		[boxPaletteVars.backgroundBodyAlt]: `var(${themeVars.lightBackgroundBodyAlt})`,
		[boxPaletteVars.backgroundShadeAlt]: `var(${themeVars.lightBackgroundShadeAlt})`,
		[boxPaletteVars.border]: `var(${themeVars.lightBorder})`,
		[boxPaletteVars.borderMuted]: `var(${themeVars.lightBorderMuted})`,
		[boxPaletteVars.accent]: `var(${themeVars.lightAccent})`,
		[boxPaletteVars.focus]: `var(${themeVars.lightFocus})`,
		[boxPaletteVars.overlay]: `var(${themeVars.lightOverlay})`,
		[boxPaletteVars.systemError]: `var(${themeVars.lightSystemError})`,
		[boxPaletteVars.systemErrorMuted]: `var(${themeVars.lightSystemErrorMuted})`,
		[boxPaletteVars.systemSuccess]: `var(${themeVars.lightSystemSuccess})`,
		[boxPaletteVars.systemSuccessMuted]: `var(${themeVars.lightSystemSuccessMuted})`,
		[boxPaletteVars.systemWarning]: `var(${themeVars.lightSystemWarning})`,
		[boxPaletteVars.systemWarningMuted]: `var(${themeVars.lightSystemWarningMuted})`,
		[boxPaletteVars.systemInfo]: `var(${themeVars.lightSystemInfo})`,
		[boxPaletteVars.systemInfoMuted]: `var(${themeVars.lightSystemInfoMuted})`,
	}),
	dark: css({
		[boxPaletteVars.palette]: 'dark',
		[boxPaletteVars.foregroundText]: `var(${themeVars.darkForegroundText})`,
		[boxPaletteVars.foregroundAction]: `var(${themeVars.darkForegroundAction})`,
		[boxPaletteVars.foregroundMuted]: `var(${themeVars.darkForegroundMuted})`,
		[boxPaletteVars.backgroundBody]: `var(${themeVars.darkBackgroundBody})`,
		[boxPaletteVars.backgroundShade]: `var(${themeVars.darkBackgroundShade})`,
		[boxPaletteVars.backgroundBodyAlt]: `var(${themeVars.darkBackgroundBodyAlt})`,
		[boxPaletteVars.backgroundShadeAlt]: `var(${themeVars.darkBackgroundShadeAlt})`,
		[boxPaletteVars.border]: `var(${themeVars.darkBorder})`,
		[boxPaletteVars.borderMuted]: `var(${themeVars.darkBorderMuted})`,
		[boxPaletteVars.accent]: `var(${themeVars.darkAccent})`,
		[boxPaletteVars.focus]: `var(${themeVars.darkFocus})`,
		[boxPaletteVars.overlay]: `var(${themeVars.darkOverlay})`,
		[boxPaletteVars.systemError]: `var(${themeVars.darkSystemError})`,
		[boxPaletteVars.systemErrorMuted]: `var(${themeVars.darkSystemErrorMuted})`,
		[boxPaletteVars.systemSuccess]: `var(${themeVars.darkSystemSuccess})`,
		[boxPaletteVars.systemSuccessMuted]: `var(${themeVars.darkSystemSuccessMuted})`,
		[boxPaletteVars.systemWarning]: `var(${themeVars.darkSystemWarning})`,
		[boxPaletteVars.systemWarningMuted]: `var(${themeVars.darkSystemWarningMuted})`,
		[boxPaletteVars.systemInfo]: `var(${themeVars.darkSystemInfo})`,
		[boxPaletteVars.systemInfoMuted]: `var(${themeVars.darkSystemInfoMuted})`,
	}),
} as const;

export type BoxPalette = keyof typeof boxPalettes;

export const boxPalette = {
	foregroundText: `var(${boxPaletteVars.foregroundText})`,
	foregroundAction: `var(${boxPaletteVars.foregroundAction})`,
	foregroundMuted: `var(${boxPaletteVars.foregroundMuted})`,
	backgroundBody: `var(${boxPaletteVars.backgroundBody})`,
	backgroundShade: `var(${boxPaletteVars.backgroundShade})`,
	backgroundBodyAlt: `var(${boxPaletteVars.backgroundBodyAlt})`,
	backgroundShadeAlt: `var(${boxPaletteVars.backgroundShadeAlt})`,
	border: `var(${boxPaletteVars.border})`,
	borderMuted: `var(${boxPaletteVars.borderMuted})`,
	accent: `var(${boxPaletteVars.accent})`,
	focus: `var(${boxPaletteVars.focus})`,
	overlay: `var(${boxPaletteVars.overlay})`,
	systemError: `var(${boxPaletteVars.systemError})`,
	systemErrorMuted: `var(${boxPaletteVars.systemErrorMuted})`,
	systemSuccess: `var(${boxPaletteVars.systemSuccess})`,
	systemSuccessMuted: `var(${boxPaletteVars.systemSuccessMuted})`,
	systemWarning: `var(${boxPaletteVars.systemWarning})`,
	systemWarningMuted: `var(${boxPaletteVars.systemWarningMuted})`,
	systemInfo: `var(${boxPaletteVars.systemInfo})`,
	systemInfoMuted: `var(${boxPaletteVars.systemInfoMuted})`,
};

/**
 * Returns the current palette for a specific DOM element
 * Note: As this function relies on CSS vars, the value returned will not be available on the server
 */
export function useBoxPalette(element: RefObject<HTMLElement>) {
	const [value, setValue] = useState<BoxPalette>();

	useEffect(() => {
		if (!element.current) return;
		setValue(getInheritedProperty(element.current));
	}, [element]);

	return value;
}

function getInheritedProperty(el: HTMLElement): BoxPalette | undefined {
	const value = getComputedStyle(el).getPropertyValue(boxPaletteVars.palette);
	if (value === 'light') return 'light';
	if (value === 'dark') return 'dark';
	// Walk up the DOM tree until we find the closest value
	if (el.parentElement) return getInheritedProperty(el.parentElement);
	return undefined;
}
