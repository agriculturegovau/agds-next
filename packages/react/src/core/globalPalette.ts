import { themeVars } from './theme';

export const globalPalette = {
	lightForegroundText: `var(${themeVars.lightForegroundText})`,
	lightForegroundAction: `var(${themeVars.lightForegroundAction})`,
	lightForegroundFocus: `var(${themeVars.lightForegroundFocus})`,
	lightForegroundMuted: `var(${themeVars.lightForegroundMuted})`,
	lightBackgroundBody: `var(${themeVars.lightBackgroundBody})`,
	lightBackgroundShade: `var(${themeVars.lightBackgroundShade})`,
	lightBackgroundBodyAlt: `var(${themeVars.lightBackgroundBodyAlt})`,
	lightBackgroundShadeAlt: `var(${themeVars.lightBackgroundShadeAlt})`,
	lightSelected: `var(${themeVars.lightSelected})`,
	lightSelectedMuted: `var(${themeVars.lightSelectedMuted})`,
	lightBorder: `var(${themeVars.lightBorder})`,
	lightAccent: `var(${themeVars.lightAccent}`,
	lightOverlay: `var(${themeVars.lightOverlay}`,
	lightOverlayMuted: `var(${themeVars.lightOverlayMuted}`,
	lightSystemSuccess: `var(${themeVars.lightSystemSuccess})`,
	lightSystemSuccessMuted: `var(${themeVars.lightSystemSuccessMuted})`,
	lightSystemError: `var(${themeVars.lightSystemError})`,
	lightSystemErrorMuted: `var(${themeVars.lightSystemErrorMuted})`,
	lightSystemWarning: `var(${themeVars.lightSystemWarning})`,
	lightSystemWarningMuted: `var(${themeVars.lightSystemWarningMuted})`,
	lightSystemInfo: `var(${themeVars.lightSystemInfo})`,
	lightSystemInfoMuted: `var(${themeVars.lightSystemInfoMuted})`,
	darkForegroundText: `var(${themeVars.darkForegroundText})`,
	darkForegroundAction: `var(${themeVars.darkForegroundAction})`,
	darkForegroundFocus: `var(${themeVars.darkForegroundFocus})`,
	darkForegroundMuted: `var(${themeVars.darkForegroundMuted})`,
	darkBackgroundBody: `var(${themeVars.darkBackgroundBody})`,
	darkBackgroundShade: `var(${themeVars.darkBackgroundShade})`,
	darkBackgroundBodyAlt: `var(${themeVars.darkBackgroundBodyAlt})`,
	darkBackgroundShadeAlt: `var(${themeVars.darkBackgroundShadeAlt})`,
	darkSelected: `var(${themeVars.darkSelected})`,
	darkSelectedMuted: `var(${themeVars.darkSelectedMuted})`,
	darkBorder: `var(${themeVars.darkBorder})`,
	darkAccent: `var(${themeVars.darkAccent}`,
	darkOverlay: `var(${themeVars.darkOverlay}`,
	darkOverlayMuted: `var(${themeVars.darkOverlayMuted}`,
	darkSystemSuccess: `var(${themeVars.darkSystemSuccess})`,
	darkSystemSuccessMuted: `var(${themeVars.darkSystemSuccessMuted})`,
	darkSystemError: `var(${themeVars.darkSystemError})`,
	darkSystemErrorMuted: `var(${themeVars.darkSystemErrorMuted})`,
	darkSystemWarning: `var(${themeVars.darkSystemWarning})`,
	darkSystemWarningMuted: `var(${themeVars.darkSystemWarningMuted})`,
	darkSystemInfo: `var(${themeVars.darkSystemInfo})`,
	darkSystemInfoMuted: `var(${themeVars.darkSystemInfoMuted})`,
} as const;
