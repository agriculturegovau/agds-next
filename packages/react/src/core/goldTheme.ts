import { Theme } from './theme';

// Default theme from GOLD.
export const goldTheme: Theme = {
	lightForegroundText: '#313131',
	lightForegroundAction: '#00698f',
	lightForegroundFocus: '#9263de',
	lightForegroundInverse: '#FFFFFF',
	lightForegroundMuted: '#626262',
	lightBackgroundBody: '#FFFFFF',
	lightBackgroundShade: '#f5f5f5',
	lightBackgroundBodyAlt: '#ebebeb',
	lightBackgroundShadeAlt: '#e0e0e0',
	lightBorder: '#808080',
	lightBorderMuted: '#D3D3D3',
	lightAccent: '#00698f',
	lightOverlay: 'rgba(0, 0, 0, 0.8)',
	lightSystemError: '#d60000',
	lightSystemErrorMuted: '#fdf2f2',
	lightSystemSuccess: '#0b996c',
	lightSystemSuccessMuted: '#f3faf8',
	lightSystemWarning: '#f69900',
	lightSystemWarningMuted: '#fffaf2',
	lightSystemInfo: '#00bfe9',
	lightSystemInfoMuted: '#f2fcfe',
	darkForegroundText: '#FFFFFF',
	darkForegroundAction: '#61daff',
	darkForegroundFocus: '#c390f9',
	darkForegroundMuted: '#bdd2d7',
	darkForegroundInverse: '#135e70',
	darkBackgroundBody: '#135e70',
	darkBackgroundShade: '#104f5f',
	darkBackgroundBodyAlt: '#0d414d',
	darkBackgroundShadeAlt: '#0a323c',
	darkBorder: '#95b7bf',
	darkBorderMuted: '#4D7680',
	darkAccent: '#61daff',
	darkOverlay: 'rgba(0, 0, 0, 0.8)',
	darkSystemError: '#d60000',
	darkSystemErrorMuted: '#422F47',
	darkSystemSuccess: '#0b996c',
	darkSystemSuccessMuted: '#133A4B',
	darkSystemWarning: '#f69900',
	darkSystemWarningMuted: '#393942',
	darkSystemInfo: '#00bfe9',
	darkSystemInfoMuted: '#193A5B',
} as const;
