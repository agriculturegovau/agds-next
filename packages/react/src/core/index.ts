export * from './Core';
export * from './CoreProvider';
export * from './tokens';
export * from './packs';
export * from './utils';
export * from './theme';
export * from './goldTheme';
export * from './boxPalette';
export * from './actionPalette';
export * from './globalPalette';
export * from './responsive';
