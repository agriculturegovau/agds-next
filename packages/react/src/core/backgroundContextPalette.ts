import { boxPalette } from './boxPalette';

// Background Palette

export const backgroundContextPaletteVars = {
	current: '--agds-background-context-background',
	shade: '--agds-background-context-shade',
	alt: '--agds-background-context-alt',
};

export const backgroundContextPalettes = {
	body: {
		[backgroundContextPaletteVars.current]: boxPalette.backgroundBody,
		[backgroundContextPaletteVars.shade]: boxPalette.backgroundShade,
		[backgroundContextPaletteVars.alt]: boxPalette.backgroundBodyAlt,
	},
	bodyAlt: {
		[backgroundContextPaletteVars.current]: boxPalette.backgroundBodyAlt,
		[backgroundContextPaletteVars.shade]: boxPalette.backgroundShadeAlt,
		[backgroundContextPaletteVars.alt]: boxPalette.backgroundShadeAlt,
	},
	shade: {
		[backgroundContextPaletteVars.current]: boxPalette.backgroundShade,
		[backgroundContextPaletteVars.shade]: boxPalette.backgroundBody,
		[backgroundContextPaletteVars.alt]: boxPalette.backgroundBody,
	},
	shadeAlt: {
		[backgroundContextPaletteVars.current]: boxPalette.backgroundShadeAlt,
		[backgroundContextPaletteVars.shade]: boxPalette.backgroundBodyAlt,
		[backgroundContextPaletteVars.alt]: boxPalette.backgroundBodyAlt,
	},
} as const;

export const backgroundContextPalette = {
	current: `var(${backgroundContextPaletteVars.current})`,
	shade: `var(${backgroundContextPaletteVars.shade})`,
	alt: `var(${backgroundContextPaletteVars.alt})`,
} as const;
