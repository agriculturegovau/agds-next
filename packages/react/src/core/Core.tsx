import { PropsWithChildren } from 'react';
import { Global } from '@emotion/react';
import { goldTheme } from './goldTheme';
import { CoreProvider, CoreProviderProps } from './CoreProvider';
import { mergeTheme, Theme } from './theme';
import { boxPalettes, boxPalette } from './boxPalette';
import { tokens } from './tokens';
import { generateFontGrid } from './utils/fontGrid';
import { printTheme } from './printTheme';

export type CoreProps = PropsWithChildren<
	{
		/** If true, a CSS reset will be applied. */
		applyReset?: boolean;
		/** Enables overwriting of the default theme. */
		theme?: Theme;
	} & CoreProviderProps
>;

export function Core({
	children,
	applyReset = true,
	theme,
	linkComponent,
}: CoreProps) {
	return (
		<CoreProvider linkComponent={linkComponent}>
			<Global
				styles={[
					{
						':root': {
							...mergeTheme(goldTheme, theme),
							...generateFontGrid(),
						},
						// Reset the global theme in print mode to black & white
						// Note: Components can also contain print specific styles
						'@media print': {
							':root': mergeTheme(goldTheme, printTheme),
						},
					},
					{
						// Apply the light palette by default
						'body, html': boxPalettes.light,
					},
					applyReset && {
						'body, html': {
							margin: 0,
							padding: 0,
							background: boxPalette.backgroundBody,
							fontFamily: tokens.font.body,
						},
						html: {
							// Prevent shifting layout between short & long pages
							scrollbarGutter: 'stable',
						},
					},
				]}
			/>
			{children}
		</CoreProvider>
	);
}
