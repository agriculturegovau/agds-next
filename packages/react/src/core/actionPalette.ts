import { css } from '@emotion/react';
import { boxPalette } from '../core';

const actionPaletteVars = {
	action: '--agds-action-palette',
} as const;

export const actionPalette = {
	action: `var(${actionPaletteVars.action})`,
} as const;

export const actionPalettes = {
	base: css({
		[actionPaletteVars.action]: boxPalette.foregroundAction,
	}),
	hover: css({
		[actionPaletteVars.action]: boxPalette.foregroundAction,
	}),
};
