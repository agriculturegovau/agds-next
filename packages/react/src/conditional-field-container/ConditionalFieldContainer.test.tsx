import '@testing-library/jest-dom';
import 'html-validate/jest';
import { axe, toHaveNoViolations } from 'jest-axe';
import {
	act,
	cleanup,
	fireEvent,
	render,
	screen,
} from '../../../../test-utils';
import { useToggleState } from '../core';
import {
	ConditionalFieldContainer,
	type ConditionalFieldContainerProps,
} from './ConditionalFieldContainer';

expect.extend(toHaveNoViolations);

afterEach(cleanup);

const childTestID = 'child-element';
const buttonLabel = 'Toggle visibility';

const TestComponent = ({ visible = false }: ConditionalFieldContainerProps) => {
	const [isVisible, toggleIsVisible] = useToggleState<boolean>(
		visible,
		!visible
	);

	return (
		<>
			<button onClick={toggleIsVisible} type="button">
				{buttonLabel}
			</button>
			<ConditionalFieldContainer visible={isVisible}>
				<p data-testid={childTestID}>conditional-content</p>
			</ConditionalFieldContainer>
		</>
	);
};

function renderConditionalFieldContainer(
	props: ConditionalFieldContainerProps
) {
	return render(<TestComponent {...props} />);
}

describe('ConditionalFieldContainer', () => {
	it('renders correctly', () => {
		const { container } = renderConditionalFieldContainer({ visible: true });
		expect(container).toMatchSnapshot();
	});

	it('renders valid HTML with no a11y violations', async () => {
		const { container } = renderConditionalFieldContainer({ visible: true });
		expect(container).toHTMLValidate({
			extends: ['html-validate:recommended'],
		});
		expect(await axe(container)).toHaveNoViolations();
	});

	it('does not render the children elements when the prop `visible` is set to `undefined`', () => {
		renderConditionalFieldContainer({ visible: undefined });

		const childElement = screen.queryByTestId(childTestID);
		expect(childElement).toBe(null);
	});

	it('renders the children elements when the prop `visible` is changed from "false" to "true"', async () => {
		renderConditionalFieldContainer({ visible: false });
		const initialElement = screen.queryByTestId(childTestID);
		expect(initialElement).toBe(null);

		const buttonElement = screen.getByText(buttonLabel);
		act(() => {
			fireEvent.click(buttonElement);
		});

		const childElement = await screen.findByTestId(childTestID);
		expect(childElement).toBeInTheDocument();
	});
});
