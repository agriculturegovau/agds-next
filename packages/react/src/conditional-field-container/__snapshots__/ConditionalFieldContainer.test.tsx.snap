// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConditionalFieldContainer renders correctly 1`] = `
<div>
  <button
    type="button"
  >
    Toggle visibility
  </button>
  <div
    class="css-2j0r52-boxStyles"
    data-conditional-field-container="true"
  >
    <p
      data-testid="child-element"
    >
      conditional-content
    </p>
  </div>
</div>
`;
